nil/services/synccommittee/prover/tracer/types.go

```diff
@@ -31,50 +19,68 @@ type executionTracesImpl struct {
 	ContractsBytecode map[types.Address][]byte
 }
 
-var _ ExecutionTraces = new(executionTracesImpl)
-
-func NewExecutionTraces() ExecutionTraces {
-	return &executionTracesImpl{
+func NewExecutionTraces() *ExecutionTraces {
+	return &ExecutionTraces{
 		ContractsBytecode: make(map[types.Address][]byte),
 	}
 }
 
-func (tr *executionTracesImpl) AddMemoryOps(ops []MemoryOp) {
+func (tr *ExecutionTraces) AddMemoryOps(ops []MemoryOp) {
 	tr.MemoryOps = append(tr.MemoryOps, ops...)
 }
 
-func (tr *executionTracesImpl) AddStackOps(ops []StackOp) {
+func (tr *ExecutionTraces) AddStackOps(ops []StackOp) {
 	tr.StackOps = append(tr.StackOps, ops...)
 }
 
-func (tr *executionTracesImpl) AddStorageOps(ops []StorageOp) {
+func (tr *ExecutionTraces) AddStorageOps(ops []StorageOp) {
 	tr.StorageOps = append(tr.StorageOps, ops...)
 }
 
-func (tr *executionTracesImpl) AddZKEVMStates(states []ZKEVMState) {
+func (tr *ExecutionTraces) AddZKEVMStates(states []ZKEVMState) {
 	tr.ZKEVMStates = append(tr.ZKEVMStates, states...)
 }
 
-func (tr *executionTracesImpl) AddCopyEvents(events []CopyEvent) {
+func (tr *ExecutionTraces) AddCopyEvents(events []CopyEvent) {
 	tr.CopyEvents = append(tr.CopyEvents, events...)
 }
 
-func (tr *executionTracesImpl) AddContractBytecode(addr types.Address, code []byte) {
+func (tr *ExecutionTraces) AddContractBytecode(addr types.Address, code []byte) {
 	tr.ContractsBytecode[addr] = code
 }
 
-func (tr *executionTracesImpl) AddExpOps(ops []ExpOp) {
+func (tr *ExecutionTraces) AddExpOps(ops []ExpOp) {
 	tr.ExpOps = append(tr.ExpOps, ops...)
 }
 
-func (tr *executionTracesImpl) AddKeccakOps(ops []KeccakBuffer) {
+func (tr *ExecutionTraces) AddKeccakOps(ops []KeccakBuffer) {
 	tr.KeccakTraces = append(tr.KeccakTraces, ops...)
 }
 
-func (tr *executionTracesImpl) SetMptTraces(mptTraces *mpttracer.MPTTraces) {
+func (tr *ExecutionTraces) SetMptTraces(mptTraces *mpttracer.MPTTraces) {
 	tr.MPTTraces = mptTraces
 }
 
+// Append adds `other` to the end of traces slices, adds kv pairs from `otherTrace` maps
+func (tr *ExecutionTraces) Append(other *ExecutionTraces) {
+	if tr.MPTTraces != nil {
+		panic("you should not merge MPT traces, call `SetMptTraces` once at the end")
+	}
+	tr.MPTTraces = other.MPTTraces
```

Comment:
It's merge MPT traces or replace?

Reply:
It is the way to ensure we don't miss any MPT traces (`SetMptTraces` is called at the end, not in the middle of tracing), and that containing traces for 
Not clear how MPT traces will look at final version, I decided not to spend time on merging MPT traces from different blocks. Perhaps we will abandon the idea of having a single MPT trace for multiple blocks in favor of per-block traces.